appscale/tools/agents/azure_agent.py

```diff
@@ -700,81 +700,148 @@ def terminate_instances(self, parameters):
     resource group.
     Args:
         compute_client: An instance of the Compute Management client.
-        resource_group: The resource group name to use for this deployment.
-        verbose: A boolean indicating whether or not the verbose mode is on.
-        scale_set_name: The name of the virtual machine scale set to be deleted.
+        parameters: A dict, containing all the parameters necessary to
+          authenticate this user with Azure.
+        vmss_name: The name of the virtual machine scale set to be deleted.
     """
-    AppScaleLogger.verbose("Deleting Scale Set {} ...".format(scale_set_name), verbose)
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
+    verbose = parameters[self.PARAM_VERBOSE]
+    AppScaleLogger.verbose("Deleting Scale Set {} ...".format(vmss_name), verbose)
     try:
       delete_response = compute_client.virtual_machine_scale_sets.delete(
-        resource_group,scale_set_name)
-      resource_name = 'Virtual Machine Scale Set' + ":" + scale_set_name
+        resource_group, vmss_name)
+      resource_name = 'Virtual Machine Scale Set' + ":" + vmss_name
       self.sleep_until_delete_operation_done(delete_response, resource_name,
                                              self.MAX_VM_UPDATE_TIME, verbose)
       AppScaleLogger.verbose("Virtual Machine Scale Set {} has been successfully "
-                             "deleted.".format(scale_set_name), verbose)
+                             "deleted.".format(vmss_name), verbose)
     except CloudError as error:
       raise AgentConfigurationException("There was a problem while deleting the "
                                         "Scale Set {0} due to the error: {1}"
-                                        .format(scale_set_name, error.message))
+                                        .format(vmss_name, error.message))
+
+  def delete_vmss_instance(self, compute_client, parameters, vmss_name, instance_id):
+    """ Deletes the specified virtual machine instance from the given Scale Set.
+    Args:
+      compute_client: An instance of the Compute Management client.
+      parameters: A dict, containing all the parameters necessary to
+        authenticate this user with Azure.
+      vmss_name: The Scale Set from which the instance needs to be deleted.
+      instance_id: The ID of the instance in the Scale Set to be deleted.
+    """
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
+    verbose = parameters[self.PARAM_VERBOSE]
+    AppScaleLogger.verbose("Deleting Virtual Machine Instance {0} from Scale "
+      "Set {1} ...".format(instance_id, vmss_name), verbose)
+    result = compute_client.virtual_machine_scale_set_vms.delete(resource_group,
+                                                                 vmss_name,
+                                                                 instance_id)
+    resource_name = 'Virtual Machine Instance ' + instance_id
```

Comment:
Not really, so I have a common method for waiting until the resources are created/deleted. So I pass the resource name so that I can print in the logs which is the resource we are waiting on. I don't see any benefit in making it a constant because we would not really use it anywhere else. Unless you see it as neater?